packages/react-compose/src/next/getSlots.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { getNativeElementProps } from '@uifabric/utilities';
+import { GenericDictionary } from './types';
+import { nullRender } from './nullRender';
+
+/**
+ * Given the state and an array of slot names, will break out `slots` and `slotProps`
+ * collections. Ensures that each slot is a renderable component, and each slotProp is
+ * a list of acceptable native properties for the given slot. Native properties are only
+ * filtered if the slot is rendered "as" a string type, and will use `getNativeElementProps`
+ * to pass along only valid props. If the slot is a non-string value, no filtering is done.
+ *
+ * @param state - State including slot definitions
+ * @param slotNames - Name of which props are slots
+ * @returns An object containing the `slots` map and `slotProps` map.
+ */
+export const getSlots = (state: GenericDictionary, slotNames: string[] | undefined) => {
```

Comment:
where did the tests go?

Reply:
I haven't added them yet, I had some tests for compose. Happy to add them if we're good with this, but i was hoping to get more eyes on the general approach before i go too deep with tests. They will get done before moving this out of /next folder.

Reply:
I did add some getSlots tests; i will add more as well for the others, but I'm eager to get the button PR going to see numbers.